scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +171,373 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
         i += 1
 
     return np.reshape(x, size1d)
+
+
+class FastNumericalInverse():
```

Comment:
I am not sure about the name. In the end it provides `ppf, rvs, qrvs`. Is numerical inverse clear enough here? Do you think that adding `Sampling` or something else in the name could help?

Or I might be missing the main use case for this class, and the name is already a good fit.

Reply:
The main use case for this class is to provide fast (approximate) numerical inverses (`ppf` methods) for distributions that don't have a fast one. One (important, but not the only) use for a fast numerical inverse is sampling. So I do think it's an appropriately descriptive name.